pkg/logs/tailers/windowsevent/tailer_test.go

```diff
@@ -267,6 +267,177 @@ func (s *ReadEventsSuite) TestBookmarkNewTailer() {
 	// if tailer started from bookmark correctly, there should only be s.numEvents
 }
 
+// TestInitialBookmarkSeeding verifies the fix for the amnesia bug:
+// When a tailer starts with no bookmark, it creates an initial bookmark
+// from the most recent event and saves it immediately, even if no events
+// are processed before shutdown.
+func (s *ReadEventsSuite) TestInitialBookmarkSeeding() {
+	config := Config{
+		ChannelPath: s.channelPath,
+	}
+
+	// Step 1: Generate N=10 initial events to establish a baseline
+	// These events ensure there's a "most recent event" for initial seeding
+	initialEvents := uint(10)
+	err := s.ti.GenerateEvents(s.eventSource, initialEvents)
+	s.Require().NoError(err)
+
+	// Step 2: Start tailer with empty bookmark
+	// This triggers createInitialBookmark() which should:
+	// 1. Query for the most recent event
+	// 2. Create a bookmark from that event
+	// 3. Send a synthetic message to persist the bookmark
+	msgChan := make(chan *message.Message, 100) // Buffer to avoid blocking
+	tailer, err := newtailer(s.ti.API(), &config, "", msgChan)
+	s.Require().NoError(err)
+
+	// Step 3: Get the initial bookmark directly from the registry
+	// The implementation now uses direct SetOffset instead of synthetic messages
+	var initialBookmark string
+
+	// Wait a moment for the tailer to initialize and set the bookmark
+	time.Sleep(100 * time.Millisecond)
+
+	// Get the bookmark directly from the mock registry
+	mockRegistry := tailer.registry.(*auditormock.Auditor)
+	initialBookmark = mockRegistry.GetOffset(tailer.Identifier())
+	s.Require().NotEmpty(initialBookmark, "Initial bookmark must not be empty")
+
+	// Verify the bookmark content
+	s.Require().Contains(initialBookmark, "RecordId=", "Bookmark should contain a RecordId")
+	s.Require().Contains(initialBookmark, s.channelPath, "Bookmark should contain the channel path")
+	s.Require().Contains(initialBookmark, "BookmarkList", "Bookmark should be valid XML")
+
+	// Log the actual registry contents for debugging
+	s.T().Logf("Registry contains bookmark for %s: %s", tailer.Identifier(), initialBookmark)
+	s.T().Logf("Full mock registry state: %+v", mockRegistry.StoredOffsets)
+
+	// Optionally drain any real events that might be processed
+	// (from the initial 10 events we generated)
+	drainTimeout := time.After(2 * time.Second)
+drainLoop:
+	for {
+		select {
+		case msg := <-msgChan:
+			// Update bookmark if we process real events
+			initialBookmark = msg.Origin.Offset
+		case <-drainTimeout:
+			break drainLoop
+		}
+	}
```

Comment:
the tailer should start at latest event so it shouldn't read any of the 10 original events, so we should not need this loop. Did you see otherwise?